src/main/java/frc/robot/subsystems/leds/Leds.java

```diff
@@ -67,31 +95,44 @@ public boolean inConeMode() {
     public void periodic() {
         if (blink) {
             if (timer.hasElapsed(blinkTime)) {
-                if (mode == Mode.ON) {
-                    mode = Mode.OFF;
+                if (mode == Mode.PRIMARY_COLOR) {
+                    mode = Mode.SECONDARY_COLOR;
                 } else {
-                    mode = Mode.ON;
+                    mode = Mode.PRIMARY_COLOR;
                 }
                 timer.reset();
             }
-        } else {
-            mode = Mode.ON;
+        } else if(mode != Mode.ELSE){
+            mode = Mode.PRIMARY_COLOR;
         }
 
-        if (mode == Mode.ON) {
+
+        if (mode == Mode.PRIMARY_COLOR) {
             for (int i = 0; i < ledBuffer.getLength(); i++) {
                 ledBuffer.setLED(i, color);
             }
             leds.setData(ledBuffer);
-        } else {
+        } else if (mode == Mode.SECONDARY_COLOR) {
             for (int i = 0; i < ledBuffer.getLength(); i++) {
-                ledBuffer.setLED(i, Color.kBlack);
+                ledBuffer.setLED(i, secondaryColor);
             }
             leds.setData(ledBuffer);
         }
+        if (isRainbow){
+            mode = Mode.ELSE;
+            System.out.println("before loop statement");
+            for (int i = 0; i < ledBuffer.getLength(); i++) {
+                ledBuffer.setHSV(i, rainbowHue, 255, 180);
+                rainbowHue += (180 / ledBuffer.getLength());
+                leds.setData(ledBuffer);
+                System.out.println("in-loop check statement");
+            rainbowHue%=180;
+            }
+            System.out.println("after loop statement");
+        }
     }
```

Comment:
Take this with a grain of salt since it's been forever since ive worked in FRC,
but shouldn't if rainbow be checked _before_ the Primary and Secondary colors?

In this scenario for 1 frame you will have a flickering of the leds.

### Suggestion: (i forgot how to actually write it in the code:
above `if(blink)`:
`if(isRainbow) {mode = Mode.ELSE}`
then you just have a simple
if(PRIMARY)
else if (SECONDARY)
else{/*ELSE*/}

Reply:
The rainbow LEDs work great so I guess if it ain't broke don't fix it

Reply:
This is a good rule to follow many times, but not always. The offseason is the best time to improve software and learn, now's the time to make such changes and improvements to the code.

Reply:
This is a good rule to follow many times, but not always. The offseason is the best time to improve software and learn, now's the time to make such changes and improvements to the code.